### ISSUE TITLE
Unstable test found in CI TestInitConnect

### ISSUE THREAD

**Issue by tiancaiamao** [pingcap/tidb#30249]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2021-11-30T02:29:26.743Z] --- FAIL: TestInitConnect (1.36s)
[2021-11-30T02:29:26.743Z]     dbtestkit.go:72:
[2021-11-30T02:29:26.743Z]              Error Trace:    dbtestkit.go:72
[2021-11-30T02:29:26.743Z]                                                      server_test.go:1939
[2021-11-30T02:29:26.743Z]                                                      server_test.go:117
[2021-11-30T02:29:26.743Z]                                                      server_test.go:1938
[2021-11-30T02:29:26.743Z]                                                      tidb_test.go:1107
[2021-11-30T02:29:26.743Z]              Error:          Received unexpected error:
[2021-11-30T02:29:26.743Z]                              commands out of sync. Did you run multiple statements at once?
[2021-11-30T02:29:26.743Z]              Test:           TestInitConnect
[2021-11-30T02:29:26.743Z]              Messages:       sql:SET GLOBAL init_connect="insert into test.ts VALUES (NOW());SET @a=1;", args:[]
[2021-11-30T02:29:26.743Z]     server_test.go:112:
[2021-11-30T02:29:26.743Z]              Error Trace:    server_test.go:112
[2021-11-30T02:29:26.743Z]                                                      panic.go:613
[2021-11-30T02:29:26.743Z]                                                      testing.go:741
[2021-11-30T02:29:26.743Z]                                                      dbtestkit.go:72
[2021-11-30T02:29:26.743Z]                                                      server_test.go:1939
[2021-11-30T02:29:26.743Z]                                                      server_test.go:117
[2021-11-30T02:29:26.743Z]                                                      server_test.go:1938
[2021-11-30T02:29:26.743Z]                                                      tidb_test.go:1107
[2021-11-30T02:29:26.743Z]              Error:          Received unexpected error:
[2021-11-30T02:29:26.743Z]                              bad connection
[2021-11-30T02:29:26.743Z]              Test:           TestInitConnect
```

### 2. What did you expect to see? (Required)

CI pass

### 3. What did you see instead (Required)

Sometimes fail ... I don't know how it happens

### 4. What is your TiDB version? (Required)

```
Head:     master executor: send a task with error to the resultCh when panic happen (#30214)
Merge:    origin/master executor: send a task with error to the resultCh when panic happen (#30214)
Tag:      v5.4.0-alpha (263)
```

**Comment by hawkingrei** [comment]
Closed by #30306

**Comment by hawkingrei** [comment]
we can re-open it if the error happens again.
